app/controllers/responses_controller.rb

```diff
@@ -24,19 +24,25 @@ def new
   end
 
   def create
-    @response            = Response.new(response_params)
-    @response.survey     = @survey
-    @response.ip_address = request.remote_ip
-    @response.language   = params[:locale]
-
-    respond_to do |format|
-      if @response.save
-        format.html { redirect_to root_path, notice: 'Response was successfully created.' }
-        format.json { render :show, status: :created, location: @response }
-      else
-        format.html { render :new }
-        format.json { render json: @response.errors, status: :unprocessable_entity }
+    answer_params = validate_responses(response_params)
+
+    if answer_params.present?
+      @response            = Response.new(answer_params)
+      @response.survey     = @survey
+      @response.ip_address = request.remote_ip
+      @response.language   = params[:locale]
+
+      respond_to do |format|
+        if @response.save && answer_params.present?
```

Comment:
The `answer_params.present?` is redundant here as it's already being checked in the enclosing if statement

Reply:
Thanks well spotted!